libs/langchain-google-cloud-sql-pg/src/indexes.ts

```diff
@@ -22,10 +22,92 @@ export class DistanceStrategy extends StrategyMixin {
 export const DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.COSINE_DISTANCE;
 export const DEFAULT_INDEX_NAME_SUFFIX: string = "langchainvectorindex";
 
+export abstract class BaseIndex {
+  name: string;
+  indexType: string;
+  distanceStrategy: DistanceStrategy;
+  partialIndexes: string[]
+
+  constructor(name?: string, indexType: string = "base", distanceStrategy: DistanceStrategy = DistanceStrategy.COSINE_DISTANCE, partialIndexes?: string[]) {
+    this.name = name!;
+    this.indexType = indexType;
+    this.distanceStrategy = distanceStrategy;
+    this.partialIndexes = partialIndexes!;
+  }
+
+  /**
+   * Set index query options for vector store initialization.
+   */
+  abstract indexOptions(): string;
+}
+
+export class ExactNearestNeighbor extends BaseIndex {
+  indexType: string = "exactnearestneighbor";
+
+  constructor() {
+    super()  
+  }
+
+  indexOptions(): string {
+    return this.indexType;
```

Comment:
this should throw a not implemented. once implemented the code should not need to call this when using exact nearest neighbor